### PR TITLE
Global regex patterns test against the full filepath, like glob does.

### PR DESCRIPTION
**Pull request by roylaurie**
Global regex patterns (//g) test against the full filepath, like glob does.

NPM "main" points to dist/index.js. NPM postinstall runs a build to create /dist.

### CODE REVIEW

**Comment by AkashBabu** [comment]
@roylaurie Firstly thanks for your interest in the library.

Your PR seems to contain an interesting feature, but I cannot merge it without test cases supporting your feature. So can you please add test cases.

If incase you don't find time to add tests, let me know so that I shall add them myself.

**Comment by roylaurie** [comment]
Hi @AkashBabu,

Thanks for the library. It's fast. :)

I'll be happy to add the test cases myself this weekend.

Please let me know if there are any edits to the changes that you'd like me to make as well.

Thanks again,
Roy

File: src/index.js [hunk 1]
```diff
@@ -8,11 +8,23 @@ const fs_stat = promisify(fs.stat);
...
+                result.push({ dir, file });
+            } else if (stat.isDirectory() && depth > 0) {
+                await search(path.join(dir, file), regex, depth - 1, result, concurrency);
+            }
+        } else { // scan the entire path for the regex if the pattern uses a //g
+            if (regex.test(statPath)) {
+                result.push({ dir, file });
+            }
+
+            if (stat.isDirectory() && depth > 0) {
```
Comment (AkashBabu) [inline src/index.js]: *Please try to avoid code duplication*
Duplicate code with reference to lines 17-19.

**Review by AkashBabu — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (roylaurie) [reply src/index.js]: Fixed. Refactored in latest commit.

Comment (roylaurie) [reply src/index.js]: Unit tests are also ready for review. I copied what you had and did the same for the second codepath (//g). I also added a new test that tests the ability to search in the filepath itself.

File: package.json [hunk 2]
```diff
@@ -2,8 +2,9 @@
     "name": "file-regex",
     "version": "3.0.2",
     "description": "Find files by using regex",
-    "main": "index.js",
+    "main": "dist/index.js",
     "scripts": {
+        "postinstall": "npm run build",
```
Comment (AkashBabu) [inline package.json]: `postinstall` would mean that everyone will have to install `devDependencies`!! which is an unnecessary overhead.
So manually run the build and commit the relevant files.

**Review by AkashBabu — CHANGES_REQUESTED** [review]
Just `postinstall` is an issue!
You fix it and I'm good with merging the changes

File: package.json (hunk 2 above)
Comment (roylaurie) [reply package.json]: Got it. Removed postinstall. The only thing really necessary to fix the current broken NPM package issue is the change to the "main" property.

**Review by AkashBabu — APPROVED** [review]

**Comment by AkashBabu** [comment]
Hey after running the code in my machine I found a bug. If the filename pattern is not specified in the regex, then the results contain the directory names as well.

For ex: if the pattern is `/\/dir3/g`
then the result will container `[{ dir:
     'home//file-regex/test/files',
    file: 'dir3' }]`

Anyways I'm working on the fix for this.

**Comment by AkashBabu** [comment]
Fixed the bug and published to v3.1.0

**Comment by roylaurie** [comment]
Thank you, Akash! This helps me a lot. Have a good week!

**Pull request merged**